parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoMessageConverter.java

```diff
@@ -427,6 +485,218 @@ public void addBinary(Binary binary) {
+      this.parent = parent;
+      this.logicalTypeAnnotation = logicalTypeAnnotation;
+    }
+
+    @Override
+    public void addLong(long value) {
+      LocalTime localTime;
+      switch (logicalTypeAnnotation.getUnit()) {
+        case MILLIS:
+          localTime = LocalTime.ofNanoOfDay(value * 1_000_000);
+          break;
+        case MICROS:
+          localTime = LocalTime.ofNanoOfDay(value * 1_000);
+          break;
+        case NANOS:
+          localTime = LocalTime.ofNanoOfDay(value);
+          break;
+        default:
+          throw new IllegalArgumentException("Unrecognized TimeUnit: " + logicalTypeAnnotation.getUnit());
+      }
+      com.google.type.TimeOfDay timeOfDay = com.google.type.TimeOfDay.newBuilder()
+        .setHours(localTime.getHour())
+        .setMinutes(localTime.getMinute())
+        .setSeconds(localTime.getSecond())
+        .setNanos(localTime.getNano())
+        .build();
+      parent.add(timeOfDay);
+    }
+  }
+
+  final class ProtoDoubleValueConverter extends PrimitiveConverter {
+
+    final ParentValueContainer parent;
+
+    public ProtoDoubleValueConverter(ParentValueContainer parent) {
+      this.parent = parent;
+    }
+
+    @Override
+    public void addDouble(double value) {
+      parent.add(DoubleValue.of(value));
+    }
+  }
+
+  final class ProtoFloatValueConverter extends PrimitiveConverter {
+
+    final ParentValueContainer parent;
+
+    public ProtoFloatValueConverter(ParentValueContainer parent) {
+      this.parent = parent;
+    }
+
+    @Override
+    public void addFloat(float value) {
+      parent.add(FloatValue.of(value));
+    }
+  }
+
+  final class ProtoInt64ValueConverter extends PrimitiveConverter {
+
+    final ParentValueContainer parent;
+
+    public ProtoInt64ValueConverter(ParentValueContainer parent) {
+      this.parent = parent;
+    }
+
+    @Override
+    public void addLong(long value) {
+      parent.add(Int64Value.of(value));
+    }
+  }
+
+  final class ProtoUInt64ValueConverter extends PrimitiveConverter {
+
+    final ParentValueContainer parent;
+
+    public ProtoUInt64ValueConverter(ParentValueContainer parent) {
+      this.parent = parent;
+    }
+
+    @Override
+    public void addLong(long value) {
+      parent.add(UInt64Value.of(value));
+    }
+  }
+
+  final class ProtoInt32ValueConverter extends PrimitiveConverter {
+
+    final ParentValueContainer parent;
+
+    public ProtoInt32ValueConverter(ParentValueContainer parent) {
+      this.parent = parent;
+    }
+
+    @Override
+    public void addInt(int value) {
+      parent.add(Int32Value.of(value));
+    }
+  }
+
+  final class ProtoUInt32ValueConverter extends PrimitiveConverter {
+
+    final ParentValueContainer parent;
+
+    public ProtoUInt32ValueConverter(ParentValueContainer parent) {
+      this.parent = parent;
+    }
+
+    @Override
+    public void addLong(long value) {
+      parent.add(UInt32Value.of((int) value));
```

Comment:
Can we do a safety check when converting from int to long?

Reply:
Good idea